Dev

MaxMaceman opened the pull request:
Спринт 12

MaxMaceman committed:
Не понимаю как сделать так чтобы в истории по нажатию трек поднимался вверх списка

MaxMaceman committed:
Фикс

MaxMaceman committed:
Фикс2

MaxMaceman committed:
Фикс3, теперь вся логика соответствует ТЗ

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/App.kt:
```diff
@@ -0,0 +1,43 @@
+        applyTheme()
+    }
+
+    fun getCurrentTheme(): Boolean {
+        return if (sharedPreferences.contains(IS_DARK)) {
+            sharedPreferences.getBoolean(IS_DARK, darkTheme)
+        } else {
+            (resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) == Configuration.UI_MODE_NIGHT_YES
+        }
+    }
```
Молодец, что продумал, что значения может не быть в `sharedPreferences`, хорошая работа!

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -6,13 +6,26 @@ import android.os.Bundle
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
+        val sharedPreferences = getSharedPreferences("APP_PREFERENCES", MODE_PRIVATE)
+        val isDarkTheme = sharedPreferences.getBoolean("isDarkTheme", false)
```
Сверху заведены константы, но они не использованы тут

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -6,13 +6,26 @@ import android.os.Bundle
+        val sharedPreferences = getSharedPreferences("APP_PREFERENCES", MODE_PRIVATE)
+        val isDarkTheme = sharedPreferences.getBoolean("isDarkTheme", false)
+
+        if (isDarkTheme) {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+        } else {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+        }
```
Из активити этот код можно убрать, т.к. это у тебя уже выполняется в `onCreate` в классе `App`

ArturNurtdinov reviewed (commented):
Кнопка "Очистить историю" должна находиться под списком в соответствии с макетом:
![image](https://github.com/user-attachments/assets/2b23b6f2-7e6a-4c04-8b10-ca049c90822f)
Макет:
![image](https://github.com/user-attachments/assets/e8ca21ae-4450-40b1-b501-d55a30fd90e2)

MaxMaceman committed:
Фикс4.
Кнопка теперь под списком.
Так же заметил что у меня отсутстввовал  "hideKeyboard()" при очистке "EditText" - вернул, теперь клавиатура снова скрывается при нажатии на кнопку "Очистить" (X) в поле ввода.

MaxMaceman committed:
Небольшое измение смены очередности выполнения действий при очистке поискового запроса.
